src/android/CallNotification.java

```diff
@@ -93,19 +120,58 @@ public void show() {
                     .build();
 
             // "CallStyle notifications must be for a foreground service or user initated job or use a fullScreenIntent."
-            Intent fullScreenIntent = new Intent(this.context, IncomingCallActivity.class);
-            fullScreenIntent.putExtra("pushMessagePayload", this.pushMessagePayload);
-            PendingIntent fullScreenPendingIntent = PendingIntent.getActivity(
-                    this.context, 0, fullScreenIntent,
-                    PendingIntent.FLAG_IMMUTABLE
-            );
-
-            builder.setStyle(NotificationCompat.CallStyle.forIncomingCall(callerPerson, declinePendingIntent, answerPendingIntent));
-            builder.setFullScreenIntent(fullScreenPendingIntent, true);
+
+            if (style == style.INCOMING_CALL) {
+                builder.setStyle(NotificationCompat.CallStyle.forIncomingCall(callerPerson, declinePendingIntent, answerPendingIntent));
+
+                Intent fullScreenIntent = new Intent(this.context, IncomingCallActivity.class);
+                fullScreenIntent.putExtra("pushMessagePayload", this.pushMessagePayload);
+                PendingIntent fullScreenPendingIntent = PendingIntent.getActivity(
+                        this.context, 0, fullScreenIntent,
+                        PendingIntent.FLAG_IMMUTABLE
+                );
+                builder.setFullScreenIntent(fullScreenPendingIntent, true);
+            } else if (style == style.ONGOING_CALL) {
+                builder.setStyle(NotificationCompat.CallStyle.forOngoingCall(callerPerson, hangupPendingIntent));
+
+                PackageManager packageManager = context.getPackageManager();
+
+                Class mainActivity;
+                String  packageName = context.getPackageName();
+                Intent  launchIntent = packageManager.getLaunchIntentForPackage(packageName);
+                String  className = launchIntent.getComponent().getClassName();
+                try {
+                    mainActivity = Class.forName(className);
+                } catch (ClassNotFoundException e) {
+                    throw new RuntimeException(e);
+                }
+
+                PendingIntent contentPendingIntent = PendingIntent.getActivity(
+                        context,
+                        0,
+                        launchIntent,
+                        PendingIntent.FLAG_UPDATE_CURRENT | PendingIntent.FLAG_IMMUTABLE
+                );
+
+                Intent fullScreenIntent = new Intent(this.context, mainActivity);
+                fullScreenIntent.putExtra("pushMessagePayload", this.pushMessagePayload);
+                PendingIntent fullScreenPendingIntent = PendingIntent.getActivity(
+                        this.context, 0, fullScreenIntent,
+                        PendingIntent.FLAG_IMMUTABLE
+                );
+
+                builder.setContentIntent(contentPendingIntent);
+                builder.setFullScreenIntent(fullScreenPendingIntent, true);
```

Comment:
Am experimenting with setContentIntent and/or setFullScreenIntent with the MainActivity in the case of an ongoing call notification.

Big problem am trying to solve is preventing the card style notification from overlapping the MainActivity (and the PhoneBanner.jsx from facetalk).

<img width="205" height="282" alt="Screenshot from 2025-11-14 16-05-41" src="https://github.com/user-attachments/assets/52df50fa-9087-45c3-bd9b-b29596157cc8" />

If anyone has any ideas let me know!